tests/test_validators.py

```diff
@@ -637,3 +637,17 @@ def test_validator(self, v):
     with pytest.raises(ConfigError) as exc_info:
         make_generic_validator(test_validator)
     assert ': (self, v), "self" not permitted as first argument, should be: (cls, value' in str(exc_info.value)
+
+
+def test_assert_raises_validation_error():
+    class Model(BaseModel):
+        a: str
+
+        @validator('a')
+        def check_a(cls, v):
+            assert False, 'invalid a'
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(a='snap')
+    assert exc_info.value.errors() == [{'loc': ('a',), 'msg': 'invalid a\nassert False', 'type': 'assertion_error'}]
```

Comment:
I confirmed that 1) the extra contents do not appear if not running inside pytest, and 2) it contains precisely the contents of the assert statement prior to the comma. So the following might help make it clearer for future reference:

```suggestion
    injected_by_pytest = '\nassert False'
    assert exc_info.value.errors() == [
        {'loc': ('a',), 'msg': f'invalid a' + injected_by_pytest, 'type': 'assertion_error'}
    ]
```